UnderstandingMDSAPIs.md

```diff
@@ -0,0 +1,25 @@
+# Understanding the different MDS APIs
+
+The Mobility Data Specification offers three primary Application Programming Interfaces (APIs): `agency`, `provider`, and `policy`.
+
+Both the `agency` and `provider` APIs allow agencies to gather data about vehicles and trips from a Mobility Service Provider (MSP). Although the design of the APIs differ, they allow for similar types of data gathering. A city can adopt one or both of these APIs depending on their goals, tools, and resources.
+
+|  | **Agency API** | **Provider API** |
+| --- | --- | --- |
+| **Data flow** | MSP pushes data to Agency | Agency pulls data from MSP |
+| **Key benefits/features** | - Designed for real-time data collection<br>- Agency maintains an authoritative database of information reported by all MSPs<br>- Designed to support real-time analysis and adaptive regulation<br>- Available reference implementation and data auditing / verification tools | - Easier to use<br>- Lower IT complexity<br>- Some historical data is available on request, which may reduce need for agencies to store data<br>- Many available commercial and open source analysis tools |
+| **Drawbacks** | - More complex to implement<br>- Agency must operate or procure IT systems capable of handling real-time API calls<br>- Data only published once, not available for re-ingestion<br>- Agencies must store any data needed for future analysis or reporting<br>- Fewer software vendors currently providing processing solutions | - Harder to scale as datasets get larger<br>- Agency needs to query each MSP individually<br>- Availability of all necessary historical data from MSPs is not guaranteed<br>- Not designed for real-time right-of-way management |
```

Comment:
Thanks for the feedback!

> **Harder to scale as datasets get larger**

Agreed that the stored data issue is a wash between the two APIs, since you ultimately need to store and manage whatever you might want to reference regardless of how you get it.  But the implicit assumption here is that scaling properties of event streams are better than those of extremely large batch datasets, especially given the proliferation of cloud services designed to work with large volumes of streamed data. Does that square with your experience? Or, am I inappropriately applying the lessons of my past engineering struggles here?

>**Not designed for real-time right of way management**

Good point. There's a lot of assumptions embedded in the phrase "real-time right of way management." And fundamentally, this aspect of the doc is looking at design intent/possible future features more than it is current reality.

One of the concepts discussed by LADOT when `agency` was first announced was the idea that API calls could return "regulatory result codes" (e.g. parking at this location is considered a violation or this action will exceed a cap). This obviously hasn't been implemented to date, but it was an important reason for the real-time, bi-directional design pattern.

True that /vehicles, if implemented with low-latency and a high tolerance for frequent polling, could allow a consumer to get a near real-time picture of events in the ROW. But that use would likely pose significant technical challenges for both producers and consumers. Also, the design of `provider` lacks the kind of bi-directional channel through which to send regulatory signals back to the MSP based on individual events.

So perhaps under *Drawbacks* we replace:
> Not designed for real-time right-of-way management

with

> Does not provide a continuous, real-time data stream about events in the right-of-way
> No potential mechanism for agencies to push regulatory messages to MSPs in response to specific events

And under *Key benefits / features* for `provider` we add:
> Designed to provide batch data about recent historical events in the right-of-way and a snapshot of the status of currently deployed vehicles (*coming in 0.4.1*)

More words, but unpacks the "real-time right-of-way management" concept a bit more.

Reply:
And as I re-read this, it all feels a little too technical. Welcome suggestions... I may try to rewrite this to be more accessible.